examples/neighbors/sparse-kde.py

```diff
@@ -0,0 +1,598 @@
+
+        if len(x.shape) == 1:
+            x = x[np.newaxis, :]
+        if self.period is not None:
+            xij = np.zeros(self.means.shape)
+            xij = rij(self.period, xij, x)
+        else:
+            xij = x - self.means
+        p = (
+            self.weights
+            * self.norm
+            * np.exp(
+                -0.5 * (xij[:, np.newaxis, :] @ self.cov_inv @ xij[:, :, np.newaxis])
+            ).reshape(-1)
+        )
+        sum_p = np.sum(p)
+        if i is None:
+            return sum_p
+
+        return np.sum(p[i]) / sum_p
+
+
+# %%
+def rij(period: np.ndarray, xi: np.ndarray, xj: np.ndarray) -> np.ndarray:
+    """Get the position vectors between two points. PBC are taken into account."""
+    xij = xi - xj
+    if period is not None:
+        xij -= np.round(xij / period) * period
+
+    return xij
+
+
+# %%
+# The original model that we want to fit:
+original_model = GaussianMixtureModel(np.full(3, 1 / 3), means, covariances)
+# The fitted model:
+fitted_model = GaussianMixtureModel(
+    estimator._sample_weights, estimator._grids, estimator.bandwidth_
+)
+
+# To plot the probability density contour, we need to create a grid of points:
+x, y = np.meshgrid(np.linspace(-6, 12, 100), np.linspace(-8, 8))
+points = np.concatenate(np.stack([x, y], axis=-1))
+probs = np.array([original_model(point) for point in points])
+fitted_probs = np.array([fitted_model(point) for point in points])
+
+fig, ax = plt.subplots()
+ct1 = ax.contour(x, y, probs.reshape(x.shape), colors="blue")
+ct2 = ax.contour(x, y, fitted_probs.reshape(x.shape), colors="orange")
+h1, _ = ct1.legend_elements()
+h2, _ = ct2.legend_elements()
+ax.legend(
+    [h1[0], h2[0]],
+    ["original", "fitted"],
+)
+ax.set_xlabel("x")
+ax.set_ylabel("y")
+
+# %%
+# The performance of the probability density estimation can be characterized by the
+# Mean Integrated Squared Error (MISE), which is defined as:
+# $$ \text{MISE}=\text{E}[\int (\hat{P}(\textbf{x})-P(\textbf{x}))^2 d\textbf{x}]$$
+
+# %%
+RMSE = np.sum((probs - fitted_probs) ** 2 * (x[0][1] - x[0][0]) * (y[1][0] - y[0][0]))
+print(f"RMSE = {RMSE:.2e}")
+
+# %%
+# We can compare the result with the KDE class from scikit-learn. (Usually takes
+# several minutes to run)
+
+# %%
+kde = gaussian_kde(samples.T)
+sklearn_probs = kde(np.vstack([x.ravel(), y.ravel()])).T
+RMSE_kde = np.sum(
+    (probs - sklearn_probs) ** 2 * (x[0][1] - x[0][0]) * (y[1][0] - y[0][0])
+)
+print(f"RMSE_kde = {RMSE_kde:.2e}")
+
+# %%
+# We can see that the fitted model can perfectly capture the original one. Eventhough we
+# have not specified the number of the gaussians, it can still perform well. This
+# ability enables us to analyze the distribution of the data objectively, which is
+# important and also hard to do in analyzing the molecular simulation data. Also, we
+# see that our method is slightly better than the one from scikit-learn, but within a
+# much shorter time.
```

Comment:
I don't think you have to point out the importance of density estimation. The difference in RMSE is so marginal that I am afraid this can change due to an update in numpy or scipy so I would use `comparable`. 

```suggestion
# have not specified the number of the Gaussians, it can still perform well. This
# allows us to fit distributions of the data automatically at a comparable quality
# within a much shorter time than scikit-matter
```